bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/DataFlowDiagramConverter.java

```diff
@@ -0,0 +1,432 @@
+                    if (flow.getDestinationNode() == node) {
+                        if (flow.getEntityName().equals(packet)) {
+                            inpin = flow.getDestinationPin();
+                        }
+                    }
+                }
+
+                var assignment = ddFactory.createForwardingAssignment();
+                assignment.setOutputPin(outpin);
+                assignment.getInputPins().add(inpin);
+                behavior.getAssignment().add(assignment);
+            } else if (behaviorString.contains("set ")) {
+                String regex = "\\bset\\s+(\\S+)\\s+=\\s+(.+)\\b";
+
+                Pattern pattern = Pattern.compile(regex);
+                Matcher matcher = pattern.matcher(behaviorString);
+                matcher.find();
+
+                String variable = matcher.group(1);
+                String typeName = variable.split("\\.")[0];
+                String valueName = variable.split("\\.")[1];
+
+                Label value = null;
+                for (LabelType labelType : dd.getLabelTypes()) {
+                    if (labelType.getEntityName().equals(typeName)) {
+                        for (Label label : labelType.getLabel()) {
+                            if (label.getEntityName().equals(valueName)) {
+                                value = label;
+                            }
+                        }
+                    }
+                }
+                Assignment assignment = ddFactory.createAssignment();
+
+                assignment.getInputPins().addAll(behavior.getInPin());
+                assignment.setOutputPin(outpin);
+                assignment.getOutputLabels().add(value);
+
+                behavior.getAssignment().add(assignment);
+
+                List<String> expressions = Arrays.asList(matcher.group(2).split(" "));
+                Stack<String> stack = new Stack<>();
+                stack.addAll(expressions);
+
+                Term processedTerm = null;
+                while (!stack.isEmpty()) {
+                    String current = stack.pop();
+                    if (current.equals("TRUE")) {
+                        TRUE ddTrue = ddFactory.createTRUE();
+                        if (processedTerm == null) {
+                            processedTerm = ddTrue;
+                        }
+                    } else if (current.equals("FALSE")) {
+                        NOT ddFalse = ddFactory.createNOT();
+                        ddFalse.setNegatedTerm(ddFactory.createTRUE());
+                        if (processedTerm == null) {
+                            processedTerm = ddFalse;
+                        }
+                    }
+                }
+                assignment.setTerm(processedTerm);
+            }
+        }
+    }
+
+    private void putValue(Map<Node, Map<Pin, String>> nestedHashMap, Node key1, Pin key2, String value) {
+        nestedHashMap.computeIfAbsent(key1, k -> new HashMap<>()).put(key2, value);
+    }
```

Comment:
Rename variable key1 to node and key2 to pin (or make this method type independent)